solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts

```diff
@@ -245,7 +247,7 @@ describe("RandomBeacon - Group Creation", () => {
           context("when dkg result was approved", async () => {
             beforeEach(async () => {
               await mineBlocks(params.dkgResultChallengePeriodLength)
-              await randomBeacon.approveDkgResult()
+              await randomBeacon.connect(thirdParty).approveDkgResult()
```

Comment:
I think we messed up roles in this test and it's time to clear them up. We use `thirdParty` as a role of DKG result submitter and as a role of some random address, outside of the signing group that is e.g. challenging DKG result. In fact, `thirdParty` is the signing group member and DKG result submitter so tests are checking something else than what we think they do. 

See that:
```
const signers = await registerOperators(
  contracts.randomBeacon as RandomBeacon,
  (await getUnnamedAccounts()).slice(1, 1 + constants.groupSize)
)
```

but later:
```
thirdParty = await ethers.getSigner((await getUnnamedAccounts())[1])
```

What I think should happen is:
```
const signers = await registerOperators(
  contracts.randomBeacon as RandomBeacon,
  (await getUnnamedAccounts()).slice(1, 1 + constants.groupSize)
)

// (...)

thirdParty = await ethers.getSigner((await getUnnamedAccounts())[0])
submitter = await ethers.getSigner((await getUnnamedAccounts())[1])
notSubmitter = await ethers.getSigner((await getUnnamedAccounts())[55])
```

and then we need to inspect all test cases where `thirdParty` is used and either leave it there or replace it with `submitter`. For example, in `approveDkgResult` tests, the scenario of `with challenge period passed -> called by a third party` does not really use a third party.

cc @nkuba @dimpar @lukasz-zimnoch for potential merge conflict resolution